pkg/controller/performanceprofile/components/tuned/tuned_test.go

```diff
@@ -48,47 +66,92 @@ var _ = Describe("Tuned", func() {
 		})
 
 		It("should generate yaml with expected parameters for Isolated balancing disabled", func() {
-			profile := testutils.NewPerformanceProfile("test")
-			f := false
-			profile.Spec.CPU.BalanceIsolated = &f
-			tuned, err := NewNodePerformance(testAssetsDir, profile)
-			Expect(err).ToNot(HaveOccurred())
-			y, err := yaml.Marshal(tuned)
-			Expect(err).ToNot(HaveOccurred())
-			manifest := string(y)
+			profile.Spec.CPU.BalanceIsolated = pointer.BoolPtr(false)
+			manifest := getTunedManifest(profile)
+
 			Expect(cmdlineRealtimeWithoutCPUBalancing.MatchString(manifest)).To(BeTrue())
 		})
 
 		It("should generate yaml with expected parameters for additional kernel arguments", func() {
-			profile := testutils.NewPerformanceProfile("test")
 			profile.Spec.AdditionalKernelArgs = additionalArgs
-			tuned, err := NewNodePerformance(testAssetsDir, profile)
-			Expect(err).ToNot(HaveOccurred())
-			y, err := yaml.Marshal(tuned)
-			Expect(err).ToNot(HaveOccurred())
-			manifest := string(y)
+			manifest := getTunedManifest(profile)
+
 			Expect(cmdlineAdditionalArg.MatchString(manifest)).To(BeTrue())
 		})
 
 		It("should not allocate hugepages on the specific NUMA node via kernel arguments", func() {
-			profile := testutils.NewPerformanceProfile("test")
-			dummyNode := int32(1)
-			tuned, err := NewNodePerformance(testAssetsDir, profile)
-			Expect(err).ToNot(HaveOccurred())
-			y, err := yaml.Marshal(tuned)
-			Expect(err).ToNot(HaveOccurred())
-			manifest := string(y)
+			manifest := getTunedManifest(profile)
 			Expect(strings.Count(manifest, "hugepagesz=")).Should(BeNumerically("==", 2))
 			Expect(strings.Count(manifest, "hugepages=")).Should(BeNumerically("==", 3))
-			profile.Spec.HugePages.Pages[0].Node = &dummyNode
-			tuned, err = NewNodePerformance(testAssetsDir, profile)
-			Expect(err).ToNot(HaveOccurred())
-			y, err = yaml.Marshal(tuned)
-			Expect(err).ToNot(HaveOccurred())
-			manifest = string(y)
+
+			profile.Spec.HugePages.Pages[0].Node = pointer.Int32Ptr(1)
+			manifest = getTunedManifest(profile)
 			Expect(strings.Count(manifest, "hugepagesz=")).Should(BeNumerically("==", 1))
 			Expect(strings.Count(manifest, "hugepages=")).Should(BeNumerically("==", 2))
 		})
 
+		Context("with 1G default huge pages", func() {
+			Context("with requested 2M huge pages allocation on the specified node", func() {
+				It("should append the dummy 2M huge pages kernel arguments", func() {
```

Comment:
How about verification that dummy args were removed when we get rid of 2M from profile?

Reply:
will add it

Reply:
done